### PR TITLE
Add rewrite for BITVECTOR_ITE with term_then == term_else.

### PR DESCRIPTION
**Pull request by aniemetz**

### CODE REVIEW

File: src/theory/bv/theory_bv_rewrite_rules.h [hunk 1]
```diff
@@ -430,134 +432,135 @@ class RewriteRule {
...
+  RewriteRule<ConcatFlatten>                  rule01;
+  RewriteRule<ConcatExtractMerge>             rule02;
+  RewriteRule<ConcatConstantMerge>            rule03;
+  RewriteRule<ExtractExtract>                 rule04;
+  RewriteRule<ExtractWhole>                   rule05;
+  RewriteRule<ExtractConcat>                  rule06;
+  RewriteRule<ExtractConstant>                rule07;
+  RewriteRule<FailEq>                         rule08;
+  RewriteRule<SimplifyEq>                     rule09;
+  RewriteRule<ReflexivityEq>                  rule10;
```
Comment (mpreiner) [inline src/theory/bv/theory_bv_rewrite_rules.h]: Missing space?

Comment (aniemetz) [reply src/theory/bv/theory_bv_rewrite_rules.h]: Hm, it's perfectly aligned in vim.

File: src/theory/bv/theory_bv_rewriter.cpp [hunk 2]
```diff
@@ -169,8 +169,8 @@ RewriteResponse TheoryBVRewriter::RewriteSge(TNode node, bool prerewrite){
 RewriteResponse TheoryBVRewriter::RewriteITEBv(TNode node, bool prerewrite)
 {
   Node resultNode =
-      LinearRewriteStrategy<RewriteRule<EvalITEBv>, RewriteRule<BvIte> >::apply(
-          node);
+      LinearRewriteStrategy<RewriteRule<EvalITEBv>,
+                            RewriteRule<BvIteConstCond> >::apply(node);
```
[the remark this answers is not in the document]
Comment (aniemetz) [reply src/theory/bv/theory_bv_rewriter.cpp]: Whoops. Good catch!

**Comment by aniemetz** [comment]
Addressed, PTAL.

**Pull request merged**